### ISSUE TITLE
CriticTwin class didn't use minimum of q1 and q2 as output

### ISSUE THREAD

**Issue by zzR2718** [Yonv1943/DL_RL_Zoo#3]
in AgentNet.py, in class CriticTwin, def forward(xxx):
        x = torch.cat((state, action), dim=1)
        q_value = self.net1(x)
        return q_value
 It seems the CriticTwin is using only self.net1 as output. When this class is used to initialize AgentSAC.cri, the self.cri will only output net1 result when it's called.

When we train with AgentSAC,This will have no influence in computing target, since function get__q1_q2() is called there;but when we compute actor loss, self.cri is called, then there might be an issue. I suggest to replace the above 3 lines into:
        x = torch.cat((state, action), dim=1)
        q_value1 = self.net1(x)
        q_value2 = self.net2(x)
        q12 = torch.cat((q_value1,q_value2), dim=1)
        q_value_min = torch.min(q12,dim=1,keepdim=True,out=None)[0]
        return q_value_min

I've noticed the both the current version and revised version works well with SAC agent on 'BipedalWalker-v3', 'MountainCarContinuous-v0'. I haven't tested on other envs.

**Comment by Yonv1943** [comment]
Thank you for your commented.

You said: "It seems the CriticTwin is using only self.net1 as output. "

+ In TD3, the author use only self.net1as policy gradient (actor loss). You can see it in TD3 author's implement in (https://github.com/sfujim/TD3/blob/master/TD3.py `def Q1(self,`). TD3 (actor loss q1 only) use a lower calculation but unstable way.
+In SAC, the author use min(q1, q2) as policy gradient as your suggestion. SAC (actor loss min(q1, q2)) use a slight higher calculation but stable way.

Besides, see `next_q_target = torch.min(*self.cri_target.get__q1_q2(next_s, next_a_noise))  # CriticTwin`
I think there is no need to use `q12 = torch.cat((q_value1,q_value2), dim=1)`. I use `torch.min(q1, q2)` as above

我無法寫出比較好的英文，語氣措辭可能有些言不達意，希望海涵（我看了你的github，也許你看得懂中文）

**Comment by zzR2718** [comment]
谢谢你的解释。没错，这个更改似乎不影响性能，我也测试过，你的implementation在BipedalWalker-v3的测试中收敛速度比选取最小值的版本收敛更快。
